staging/src/k8s.io/apimachinery/pkg/util/wait/wait.go

```diff
@@ -418,13 +437,47 @@ func ExponentialBackoff(backoff Backoff, condition ConditionFunc) error {
 //
 // If you want to Poll something forever, see PollInfinite.
 func Poll(interval, timeout time.Duration, condition ConditionFunc) error {
-	return pollInternal(poller(interval, timeout), condition)
+	return poll(context.Background(), false, poller(interval, timeout), condition.WithContext())
```

Comment:
Create a `PollWithContext` function and call it here please.  Understand that during this review I'm trying to find peer functions and to ensure they are in lock-step.